src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java

```diff
@@ -0,0 +1,234 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.sizes;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks lambda body length.
+ * </p>
+ * <p>
+ * Rationale: Similar to anonymous inner classes, if lambda body becomes very long
+ * it is hard to understand and to see the flow of the method
+ * where the lambda is defined. Therefore long lambda body
+ * should usually be extracted to method.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code max} - Specify the maximum number of lines allowed.
+ * Type is {@code int}.
+ * Default value is {@code 10}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check to accept lambda bodies with up to 10 lines:
+ * </p>
+ * <pre>
+ * &lt;module name="LambdaBodyLength"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test {
+ *   Runnable r = () -&gt; { // violation, 11 lines
+ *       System.out.println(2); // line 2 of lambda
```

Comment:
please confirm that we count lines with curly braces in other Checks.
Users might find it not obvious that counting is different for cases with and without curly braces.

they are visually take the same size on screen, so might have same way of counting.

Reply:
MethodLengthCheck
https://github.com/checkstyle/checkstyle/blob/3c2249b239cbfe002af3649f1f7c1f9ae61df3b1/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java#L160-L161

AnonInnerLengthCheck
https://github.com/checkstyle/checkstyle/blob/3c2249b239cbfe002af3649f1f7c1f9ae61df3b1/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/AnonInnerLengthCheck.java#L109-L110

Reply:
but all such cases cannot skip `{` and identification(name of class, method) usually stay on same line as `{`.
So we count them as they are part of what user has to read/view.

the same for lambda as arguments  `() ->` are also part that use must see/read/...